aiperf/metrics/types/good_request_count_metric.py

```diff
@@ -0,0 +1,96 @@
+# SPDX-FileCopyrightText: Copyright (c) 2025 NVIDIA CORPORATION & AFFILIATES.
+# SPDX-License-Identifier: Apache-2.0
+
+from typing import ClassVar
+
+from aiperf.common.enums import GenericMetricUnit, MetricFlags
+from aiperf.common.exceptions import MetricTypeError, NoMetricValue
+from aiperf.common.models import ParsedResponseRecord
+from aiperf.metrics.base_aggregate_counter_metric import BaseAggregateCounterMetric
+from aiperf.metrics.metric_dicts import MetricRecordDict
+from aiperf.metrics.metric_registry import MetricRegistry
+
+
+class GoodRequestCountMetric(BaseAggregateCounterMetric):
+    """
+    Counts requests that satisfy all user-provided SLO thresholds.
+    """
+
+    tag = "good_request_count"
+    header = "GoodRequestCount"
+    short_header_hide_unit = True
+    unit = GenericMetricUnit.REQUESTS
+    flags = MetricFlags.GOODPUT | MetricFlags.NO_CONSOLE
+    required_metrics: set[str] | None = None
+
+    _thresholds: ClassVar[dict[str, float]] = {}
+
+    @classmethod
+    def set_slos(cls, slos: dict[str, float] | None) -> None:
+        """
+        Configure SLO thresholds and update dependencies.
+        """
+        slos = slos or {}
+        if not slos:
+            cls._thresholds = {}
+            cls.required_metrics = None
+            return
+
+        normalized_slos: dict[str, float] = {}
+
+        for metric_tag, value in slos.items():
+            try:
+                metric_cls = MetricRegistry.get_class(metric_tag)
+            except MetricTypeError as e:
+                raise ValueError(
+                    f"Unknown metric tag(s) in --goodput: {metric_tag}."
+                ) from e
+            unit = getattr(metric_cls, "unit", None)
+            display_unit = getattr(metric_cls, "display_unit", None)
+            if display_unit is not None and unit is not None and display_unit != unit:
```

Comment:
this should be all you need to do this properly.
```suggestion
            unit = metric_cls.unit
            display_unit = metric_cls.display_unit
            if display_unit != unit:
```